Restructuring and addition to permissions docs (#3270)
hashicorp/boundary: website/content/docs/concepts/security/permissions/advanced.mdx

---
layout: docs
page_title: Advanced Concepts
description: |-

>   Advanced permissions concepts

Author: ```suggestion
  Advanced permissions concepts 
```

---

> # Advanced Concepts

Author: will add some text in between

currently supported, as we don't perform validation on the values given.
However, this means setting `output_fields=none` is functionally equivalent!)

### Anonymous user restrictions

Starting in Boundary 0.9.0, there are severe limits placed on the actions
allowed to be assigned to the anonymous user `u_anon`. This is meant to avoid
situations where an operator accidentally or mistakenly adds the anonymous user
to a role that provides more privileges than might be intended for
unauthenticated users.

The set of actions is currently restricted to listing scopes and auth methods
and authentication to auth methods (plus `no-op` actions for listing
visibility). If further use cases arise from user feedback, this list can be
expanded.

When an anonymous user is denied access, there is no special error message that is returned.
It is still possible to assign grants to
a role on which the anonymous user is a principal that can never match the
anonymous user. In the future this may change, but trying to limit this
introduces other restrictions on reasonable administrative workflows, so an
acceptable set of tradeoffs would need to be figured out.

> ## Grant Formats
> ### Wildcard

Author: ```suggestion
## Wildcard
```

> ### Wildcard

Author: yeah

Various wildcard possibilities are allowed:

> #### Wildcard ID

Author: ```suggestion
### Wildcard ID
```

When you use only a wildcard for the ID, `*`, it matches all IDs of the given type. You can use this format
with both top-level resource types or none. For example:

`id=*;type=host-set;actions=create,read,update,set-hosts`

> #### Wildcard type

Author: ```suggestion
### Wildcard type
```

You can use a wildcard for the `type` for any non-top-level resources with pinned IDs:

`id=hcst_1234567890;type=*;actions=create,read,update`

This grant format would allow `create`, `read`, and `update` actions for all types of
subordinate resources (in this case host sets and hosts) underneath the host
catalog with ID `hcst_1234567890`.

> #### Wildcard ID and type

Author: ```suggestion
### Wildcard ID and type
```

If ID and type are both a wildcard, the grant is essentially a catch-all that

> will match any resource of any type within the scope and allow the given

Reviewer: ```suggestion
matches any resource of any type within the scope and allows the given
```

actions.

`id=*;type=*;actions=read,list`

> #### Wildcard ID, Type, and Actions

Reviewer: ```suggestion
#### Wildcard ID, type, and actions
```

Finally, ID, type, and actions can all be wildcards:

`id=*;type=*;actions=*`

Such a grant is essentially a full administrator grant for a scope.

> ### Templates

Author: ```suggestion
## Templates
```

> A few template possibilities exist, which will at grant evaluation time
> substitute the given value into the ID field of the grant string:

Reviewer: ```suggestion
A few template possibilities exist, which will 
substitute the given value into the ID field of the grant string at grant evaluation time:
```
